x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileIntegTests.java

```diff
@@ -530,6 +556,21 @@ public void testSetEnabled() {
         assertThat(profile2.uid(), equalTo(profile1.uid()));
         assertThat(profile2.enabled(), is(false));
 
+        // And can also check privileges
+        ProfileHasPrivilegesResponse profileHasPrivilegesResponse = client().execute(
+            ProfileHasPrivilegesAction.INSTANCE,
+            new ProfileHasPrivilegesRequest(
+                List.of(profile1.uid()),
+                new PrivilegesToCheck(
+                    new String[] { "cluster:monitor/state" },
+                    new RoleDescriptor.IndicesPrivileges[0],
+                    new RoleDescriptor.ApplicationResourcePrivileges[0]
+                )
+            )
+        ).actionGet();
+        assertThat(profileHasPrivilegesResponse.hasPrivilegeUids(), contains(equalTo(profile1.uid())));
+        assertThat(profileHasPrivilegesResponse.errorUids(), emptyIterable());
```

Comment:
Honestly I haven't thought about this. The envisioned use case is that the UIDs from a pre-flight search request which means they are all **enabled** (because search filtered out disabled profiles). So I had always assumed the hasPrivileges would only work for enabled profiles. But even when all UIDs are from a search result, it is theoretically possible that they got disabled just before the hasPrivileges check. Should we care? 

Functional wise, whether profile is enabled or not feels orthogonal to hasPrivileges. So I am tempted to say hasPrivileges should *not* care about whether profiles are enabled. But in previous discussion to the Kibana team, we emphasized that the search result is a best effort in terms of privileges (spaces etc) and Kibana must filter them with the hasPrivileges API to be sure. That makes me think it is rather let-down if hasPrivileges API does not care about "enabled" status which can be argued to be the top level privilege. This means we cannot fully trust what hasPrivileges says because the Profile can be disabled (and thus the associated user should not recieved a notification) despite passing the privilege check.

In summary, I think we should **not** allow hasPrivileges to work for disabled profiles.